engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -291,15 +348,38 @@ object EngineJobExecutionActor {
       callCachingMode = callCachingMode)).withDispatcher(EngineDispatcher)
   }
 
-  private[execution] sealed trait EJEAData { override def toString = getClass.getSimpleName }
+  implicit class EnhancedSuccess(val response: SucceededResponse) extends AnyVal {
+    def updatedData(data: NoResponseData): SucceededResponseData = {
+      SucceededResponseData(response, data.hashes)
+    }
+  }
+
+  implicit class EnhancedNotSuccess(val response: BackendJobExecutionResponse) extends AnyVal {
+    def updatedData(data: NoResponseData): NotSucceededResponseData = {
+      NotSucceededResponseData(response, data.hashes)
+    }
+  }
+
+  private[execution] sealed trait EJEAData {
+    override def toString = getClass.getSimpleName
+  }
 
   private[execution] case object NoData extends EJEAData
-  private[execution] case class JobDescriptorData(jobDescriptor: BackendJobDescriptor, bjeaActorProps: Props) extends EJEAData
 
-  private[execution] case object EmptyPartialCompletionData extends EJEAData
-  private[execution] case class PartialCompletionDataWithSucceededResponse(response: SucceededResponse) extends EJEAData
-  private[execution] case class PartialCompletionDataWithHashes(hashes: Try[CallCacheHashes]) extends EJEAData
+  private[execution] case class NoResponseData(jobDescriptor: BackendJobDescriptor,
```

Comment:
`ResponsePendingData` maybe?  I wanted to communicate that there isn't yet a response.